### DISCUSSION TITLE
Share your ideas or suggestions here.

### DISCUSSION THREAD

**Discussion by csgoh** [csgoh/processpiper#2] [Ideas]
The project is still in its infancy. If you have any ideas, suggestions or anything you like to see this library can do, please leave your comment here!

**Comment by v3ss0n** [comment]
Any theming support?

**Reply by csgoh** [reply]
Yes, theming support will be available. 😀

**Comment by macaroni57** [comment]
The support for grouping along the y-axis is very strong. It would be nice to also have ability to group into columns (or the x-axis). I have a hybrid DFD / BPM diagram in mind where I am trying to show multiple dimensions:

- Activities 
- input/outputs (might put into system lane) using relationship to show specific attributes
- Business units 
- Customer Journey
- Role / Team / Level (Scaled Agile like Domain / Product Level / Team)

**Reply by csgoh** [reply]
It will be great if you could show me some graphical examples you have in mind.

**Reply by boshiuchen** [reply]
Adding onto this thread with some use cases in mind from a business user.
Phases: Along the X axis, be able to group processes even with different actors that occur in the same phase.
Reference: https://support.microsoft.com/en-us/office/create-a-cross-functional-flowchart-in-visio-for-the-web-3546da67-a092-431a-b47a-bbb808edf581
Sub processes: Marking processes into groups within the chart
Refer: https://edrawmax.wondershare.com/flowchart/what-is-flowchart-subprocess.html

**Comment by s-araqayyum** [comment]
Implementation for exporting in bpmn?

**Reply by csgoh** [reply]
Working in progress

**Comment by ijimin** [comment]
The likes of structurizr have images on dockerhub to easily run the tooling. I think you can even automate its generation on github ci.

**Reply by csgoh** [reply]
Hi @ijimin, I am sorry, I don't know exactly what you are trying to say. Are you asking whether ProcessPipier can be executed in a Docker image?

**Comment by orbrick-nc** [comment]
This is fantastic work.

Current there are horizontal swimlanes. Can you add support for vertical swimlanes so that the actors become the columns and the flow goes top-to-down?

**Comment by shop-coder** [comment]
can we get a key for the symbols needed for the shapes and add more then just circle, square, diamond
